src/aliceVision/localization/optimization.cpp

```diff
@@ -302,20 +300,19 @@ bool refineSequence(std::vector<LocalizationResult>& vec_localizationResult,
         // if we used the same intrinsics for all the localization results we need to
         // update the intrinsics of each localization result
 
-        // get its optimized parameters
-        std::vector<double> params = tinyScene.getIntrinsics().at(0).get()->getParams();
-        ALICEVISION_LOG_DEBUG("Type of intrinsics " << tinyScene.getIntrinsics().at(0).get()->getType());
-        if (params.size() == 4)
+        auto intrinsic = tinyScene.getIntrinsics().at(0);
+        auto casted = camera::IntrinsicScaleOffsetDisto::cast(intrinsic);
```

Comment:
```suggestion
        auto castIntrinsic = camera::IntrinsicScaleOffsetDisto::cast(intrinsic);
```

`casted` is both unexplicit and incorrect.